vllm/transformers_utils/config.py

```
@@ -927,3 +927,33 @@ def get_model_path(model: Union[str, Path], revision: Optional[str] = None):
 
     from huggingface_hub import snapshot_download
     return snapshot_download(repo_id=model, **common_kwargs)
+
+
+def get_hf_file_bytes(file_name: str,
+                      model: Union[str, Path],
+                      revision: Optional[str] = 'main') -> Optional[bytes]:
+    """Get file contents from HuggingFace repository as bytes."""
+    file_path = try_get_local_file(model=model,
+                                   file_name=file_name,
+                                   revision=revision)
+
+    if file_path is None:
+        try:
+            hf_hub_file = hf_hub_download(model,
+                                          file_name,
+                                          revision=revision,
+                                          token=_get_hf_token())
+            file_path = Path(hf_hub_file)
+        except (OSError, ValueError) as e:
+            logger.debug("Failed to download %s from HF: %s", file_name, e)
+            return None
+
+    if file_path is not None and file_path.is_file():
+        try:
+            with open(file_path, 'rb') as file:
+                return file.read()
+        except OSError as e:
+            logger.debug("Failed to read file %s: %s", file_path, e)
```

I think no need to soft fail here. Better to just raise the error

---

Thanks for the review! I've addressed all the logging improvements you suggested. Please take a look.